enclave-boundary/src/untrusted.rs

```diff
@@ -35,27 +35,45 @@ pub fn make_variable_length_ecall(
         }
 
         let mut outbuf = vec![0u8; outbuf_used];
-        match unsafe {
-            ecall_fcn(
+        unsafe {
+            let result = ecall_fcn(
                 eid,
                 &mut retval,
                 inbuf.as_ptr(),
                 inbuf.len(),
                 outbuf.as_mut_ptr(),
                 outbuf.len(),
                 &mut outbuf_used,
-                &mut outbuf_retry_id,
-            )
-        } {
-            sgx_status_t::SGX_SUCCESS => match retval {
-                sgx_status_t::SGX_ERROR_OUT_OF_MEMORY => continue,
-                sgx_status_t::SGX_SUCCESS => {
-                    outbuf.truncate(outbuf_used);
-                    break Ok(outbuf);
+                &mut outbuf_retry_id
+            );
```

Comment:
:wrench: I think you should be able to end the unsafe block here
```suggestion
            );
        }
```

Reply:
It may need to be
```rust
let result = unsafe { ecall_fcn(...) };
``` 
or similar

Reply:
Ah, yes, I like this better